nspanel_blueprint.yaml

```diff
@@ -7116,7 +7102,6 @@ action:
               - home_value01_state
               - home_value02_state
               - home_value03_state
-          - '{{ nspanel_event.page == nextion.pages.home }}'
```

Comment:
This is my only concern here.
Is there really the need to update those values in the background? What are the risks of this overloading the system if some of those values are updated all the time? let's say every second.

Reply:
For the nextion display the 115200 baud/s is 112,5 KB/s, which is plenty.
If somebody uses sensors updated every second, they are probably overloading their systems already (HA and database).
And if they are on the homepage (and they are, for most of the time) these update will happen anyways. Probably there are people who aren't using sleep or dim mode at all, and the display is on all the time.
So if the updates could overload the display, it will happen anyway, because the display shows the home screen at least 60 sec showing the home page before going to sleep. And if the system can handle the load for a minute, it will most probably handle it all the time.
But I think only time will tell, in the future if we will have performance issues, we can still change to only refresh part of the homepage in the background.

To answer your question, if it's needed to do this or not, if you don't do it, the display will always flicker. Once when nextion loads the homepage and 0.5-1s later when the value updates.
We can also create a setting for this, and if somebody's system can't handle this load, they can turn off the background updates. They will still benefit from changing the elements to global, and not have to hide them and show them again for every page load.

Btw, in my own previous nspanel project, every variable was global and their value was updated all time. It was a smaller project, but overall had more elements then this project's front page.

Reply:
> For the nextion display the 115200 baud/s is 112,5 KB/s, which is plenty.

Indeed. I don't have concerns about the capability of the system, but more about using network resources unnecessarily.

> If somebody uses sensors updated every second, they are probably overloading their systems already (HA and database).

I do! 😂
I have a template sensor which is a sum of all the power sensors at home, and I have a power sensor on each of my power plugs, so yes, this template sensor updates all the time.
I probably should have worked on that a bit more, but it's nice to view in a dashboard que quick response when a kettle gets load, etc. And that is not really a problem for everything else other than this Blueprint, as it already runs the Blueprint in every change of every monitored entity (except by entities pages), my automation runs all the time, making troubleshooting via trace or Logbook almost useless and the behavior of the automations page is quite broken due to this Blueprint updating that screen every second.
But other than this, performance on HA is fine.
 
> And if they are on the homepage (and they are, for most of the time) these update will happen anyways. Probably there are people who aren't using sleep or dim mode at all, and the display is on all the time.

Which is not always the case. I have my panel by my front door and it is sleeping most of the time. I have an automation to wake it up with movement on that area or the door opening, so most of the time nothing is shown in the panel. And nothing is navigating thru the network.

> So if the updates could overload the display, it will happen anyway, because the display shows the home screen at least 60 sec showing the home page before going to sleep. And if the system can handle the load for a minute, it will most probably handle it all the time.

Again, my concern is not on the capacity of the panel or Home Assistant, but in the unnecessary use of my Wi-Fi.
And don't take me wrong, I'm not against the solution at all. I really think this is the direction to go, but I'm just trying to think about the corner cases and prevent having to revert this in the future.

> To answer your question, if it's needed to do this or not, if you don't do it, the display will always flicker. Once when nextion loads the homepage and 0.5-1s later when the value updates.

No. The display shouldn't be flickering today, as we currently have the `show_all` service either at the beginning of the page creation or by the end, never twice.

> We can also create a setting for this, and if somebody's system can't handle this load, they can turn off the background updates. They will still benefit from changing the elements to global, and not have to hide them and show them again for every page load.

That is another good idea, but again, let's think about, as that will increase complexity a bit.

> Btw, in my own previous nspanel project, every variable was global and their value was updated all time. It was a smaller project, but overall had more elements then this project's front page.

Yeah, you are probably right and I may be over thinking this. My concern is only about the 3 custom selected entities in the Home page (and perhaps that should extend to the chips), but that might not be a problem for this project.
I think it will be always fine if we find a user complaining about too few unnecessary updates we can always ask them to not use that kind of sensor in the Home page.

Reply:
I think the benefits here are much better than the concerns I have. We really have to rework this home page to load much faster, and what you have done is almost all the way done. 😉